Cannot authenticate with .npmrc in repository with pnpm publish 6.7.1

### pnpm version: 6.7.1
We have a monorepo with a private npm registry (Azure Artifacts) that we publish to. We noticed that publishing started failing when we updated to pnpm version 6.7.1 and we managed to resolve the issue by reverting to the previous version. (6.6.2). We use an Azure Pipeline where we use the [npm authenticate](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/package/npm-authenticate?view=azure-devops) which updates the .npmrc in the downloaded repository that we try to publish from.

It seems to exhibit the same behaviour if I do it locally, by having the authentication in the .npmrc file in the root of the repository.


### Code to reproduce the issue:

I'm sure the --no-git-checks isn't necessary, but that's the command we run.

```bash
pnpm -r publish --no-git-checks
```

### Expected behavior:
Publishing to pass as in version 6.6.2

### Actual behavior:
```bash
npm ERR! code E401
npm ERR! Unable to authenticate, need: Bearer authorization_uri=[AUTHORIZATION_URI], Basic realm="[REALM]", TFS-Federated
```

### Additional information:

npm@6.14.11
node@v14.16.0

Our .npmrc file that gets the authentication appended to it looks like the following
```
registry=[PRIVATE REGISTRY]
always-auth=true
engine-strict=true
recursive-install=true
shell-emulator=true
strict-peer-dependencies=true
```

--- a/.changeset/rare-pianos-buy.md
+++ b/.changeset/rare-pianos-buy.md
@@ -1,0 +1,5 @@
+---
+"@pnpm/plugin-commands-publishing": patch
+---
+
+Copy the `.npmrc` from the root of the repository.

--- a/packages/plugin-commands-publishing/src/publish.ts
+++ b/packages/plugin-commands-publishing/src/publish.ts
@@ -1,4 +1,4 @@
-import { promises as fs } from 'fs'
+import { promises as fs, existsSync } from 'fs'
 import path from 'path'
 import { docsUrl, readProjectManifest } from '@pnpm/cli-utils'
 import { Config, types as allTypes } from '@pnpm/config'
@@ -193,10 +193,18 @@ Do you want to continue?`,
       }
 
       const cwd = manifest.publishConfig?.directory ? path.join(dir, manifest.publishConfig.directory) : dir
+      const localNpmrc = path.join(cwd, '.npmrc')
+      const copyNpmrc = !existsSync(localNpmrc) && opts.workspaceDir && existsSync(path.join(opts.workspaceDir, '.npmrc'))
+      if (copyNpmrc && opts.workspaceDir) {
+        await fs.copyFile(path.join(opts.workspaceDir, '.npmrc'), localNpmrc)
+      }
 
       const { status } = runNpm(opts.npmPath, ['publish', '--ignore-scripts', ...args], {
         cwd,
       })
+      if (copyNpmrc) {
+        await rimraf(localNpmrc)
+      }
 
       _status = status!
     }

--- a/packages/plugin-commands-publishing/test/recursivePublish.ts
+++ b/packages/plugin-commands-publishing/test/recursivePublish.ts
@@ -67,8 +67,7 @@ test('recursive publish', async () => {
     },
   ])
 
-  await fs.writeFile(path.join(pkg1.name, '.npmrc'), CREDENTIALS, 'utf8')
-  await fs.writeFile(path.join(pkg2.name, '.npmrc'), CREDENTIALS, 'utf8')
+  await fs.writeFile('.npmrc', CREDENTIALS, 'utf8')
 
   await publish.handler({
     ...DEFAULT_OPTS,
@@ -250,8 +249,7 @@ test('recursive publish writes publish summary', async () => {
     },
   ])
 
-  await fs.writeFile('@pnpmtest/test-recursive-publish-project-3/.npmrc', CREDENTIALS, 'utf8')
-  await fs.writeFile('@pnpmtest/test-recursive-publish-project-4/.npmrc', CREDENTIALS, 'utf8')
+  await fs.writeFile('.npmrc', CREDENTIALS, 'utf8')
 
   process.env.npm_config_userconfig = path.join('.npmrc')
   await publish.handler({

